### PR TITLE
add error logging wraps for CloseHandle and LocalFree

### PR DESCRIPTION
**Pull request by Shrini-h**
#### Describe Bug or Feature

No error logs printed if CloseHandle or LocalFree fails

#### Describe Your Change

adding new wrapper functions that log the error in case of failure.
wrapper is invoked using a macro overriding all the API invocations

#### Attach Test and Valgrind Logs/Output

[20200511.txt](https://github.com/PBSPro/pbspro/files/4609332/20200511.txt)

error was forced using a debugger
```
05/11/2020 04:52:31;0100;pbs_mom;Req;;Type 1 request received from root@192.168.153.132:15001, sock=2
05/11/2020 04:52:31;0100;pbs_mom;Req;;Type 5 request received from root@192.168.153.132:15001, sock=2
05/11/2020 04:55:04;0001;pbs_mom;n/a;has_privilege;CloseHandle(procToken) at line 2810 failed with error 0, Handle[638]
05/11/2020 04:55:50;0001;pbs_mom;n/a;getdefgrpname;LocalFree(sid) at line 1923 failed with error 122, Handle[94e4728]
05/11/2020 04:56:55;0001;pbs_mom;n/a;LogonUserNoPass;CloseHandle(hToken) at line 3901 failed with error 0, Handle[640]
05/11/2020 04:57:51;0100;pbs_mom;Job;0.mixlin;allowed pbsuser to access window station and desktop, 

```

### CODE REVIEW

File: src/lib/Libwin/util.c [hunk 1]
```diff
@@ -978,3 +978,59 @@ revert_impersonated_user()
...
+ *  	A wrapper to LocalFree()
+ *  	If LocalFree fails, log the error.
+ *
+ * @param[in]   hObject : handle to be closed
+ * @param[in]   handlename : name of param containing handle
+ * @param[in]   funcname : function name where API was invoked
+ * @param[in]   lineno : line number of API invocation
+ *
+ * @return     BOOL
+ * @retval     whether CloseHandle succeded, return value of CloseHandle() API
```
Comment (SudeshnaMoulik) [inline src/lib/Libwin/util.c:1020]: One minor comment on re-wording above statement. Rest looks good to me.

Comment (Shrini-h) [reply src/lib/Libwin/util.c]: fixed in https://github.com/PBSPro/pbspro/pull/1727/commits/c21581adc9d84b34c0693da850280573508b7c6e

**Comment by Shrini-h** [comment]
@SudeshnaMoulik , thank you for the review, I have addressed your comments and attached logs where I forced some errors using a debugger

File: src/lib/Libwin/util.c [hunk 2]
```diff
@@ -978,3 +978,61 @@ revert_impersonated_user()
...
+_log_wrap_CloseHandle(HANDLE hObject, LPCSTR handlename, LPCSTR funcname, INT lineno)
+{
+	BOOL rc;
+
+#undef CloseHandle
+	rc = CloseHandle(hObject);
+#define CloseHandle(param) _log_wrap_CloseHandle(param, #param, __func__, __LINE__)
+
+	if (!rc) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_ERR, funcname,
```
Comment (subhasisb) [inline src/lib/Libwin/util.c:1004]: DO you want to add server eventclass instead of 0 here?

File: src/lib/Libwin/util.c [hunk 3]
```diff
@@ -978,3 +978,61 @@ revert_impersonated_user()
...
+_log_wrap_LocalFree(HLOCAL hObject, LPCSTR handlename, LPCSTR funcname, INT lineno)
+{
+	HLOCAL hret;
+
+#undef LocalFree
+	hret = LocalFree(hObject);
+#define LocalFree(param) _log_wrap_LocalFree(param, #param, __func__, __LINE__)
+
+	if (hObject && (hObject == hret)) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_ERR, funcname,
```
Comment (subhasisb) [inline src/lib/Libwin/util.c:1033]: same here add the server event class

File: src/lib/Libwin/util.c (hunk 2 above)
Comment (Shrini-h) [reply src/lib/Libwin/util.c]: fixed in https://github.com/PBSPro/pbspro/pull/1727/commits/d40ea6a01f1c873f0de3a7e67671b33fec9927cc

File: src/lib/Libwin/util.c (hunk 3 above)
Comment (Shrini-h) [reply src/lib/Libwin/util.c]: https://github.com/PBSPro/pbspro/pull/1727/commits/d40ea6a01f1c873f0de3a7e67671b33fec9927cc

**Comment by Shrini-h** [comment]
@subhasisb and @SudeshnaMoulik  do you think I should wait for merge of log_errf() api https://github.com/PBSPro/pbspro/pull/1751 ?

**Comment by subhasisb** [comment]
> @subhasisb and @SudeshnaMoulik do you think I should wait for merge of log_errf() api #1751 ?

That might be nicer, let's wait for it - yes?

File: src/lib/Libwin/util.c [hunk 4]
```diff
@@ -978,3 +978,61 @@ revert_impersonated_user()
...
+_log_wrap_LocalFree(HLOCAL hObject, LPCSTR handlename, LPCSTR funcname, INT lineno)
+{
+	HLOCAL hret;
+
+#undef LocalFree
+	hret = LocalFree(hObject);
+#define LocalFree(param) _log_wrap_LocalFree(param, #param, __func__, __LINE__)
+
+	if (hObject && (hObject == hret)) {
+		log_eventf(PBSEVENT_ERROR, PBS_EVENTCLASS_SERVER, LOG_ERR, funcname,
```
Comment (SudeshnaMoulik) [inline src/lib/Libwin/util.c:1033]: Will it be better if we modify these log_eventf invocations to log_errf ?
https://github.com/PBSPro/pbspro/pull/1727/files#diff-714da0a1389ce6a85c4daf0d236cdac3R1004
https://github.com/PBSPro/pbspro/pull/1727/files#diff-714da0a1389ce6a85c4daf0d236cdac3R1033

Comment (Shrini-h) [reply src/lib/Libwin/util.c]: off course yes. changed in https://github.com/PBSPro/pbspro/pull/1727/commits/1b8fecfd5b582162c3bac162c5ea63101542c851

**Pull request merged**